### ISSUE TITLE
Feature: Specify schema and table hints for all tables in query

### ISSUE THREAD

**Issue by arwyl** [linq2db/linq2db#1545]
As proposed by @sdanyliv in #1464 

```
var query = from e in db.Entities
  from e2 in db.Entities
  select new {e, e2};

var query = query.WithTableHints("NOLOCK");
var materialized = query.ToArray();
```

and also specifying Schema globally for dataconnection would be great

**Comment by sdanyliv** [comment]
Specifying schema globally will be possible when we finish new options mechanism.

**Comment by arwyl** [comment]
@sdanyliv awesome!

**Comment by MaceWindu** [comment]
Implemented by #3297 (will be released with next v4 preview)
